download_py_package.py

```diff
@@ -55,20 +55,36 @@ def download_py_package():
             version_exists = True
     else:
         if version_exists:
-            message_contents = provides_source.MessageContents(
-                package_name=Configuration.PACKAGE_NAME,
-                package_version=Configuration.PACKAGE_VERSION,
-                index_url=Configuration.PACKAGE_INDEX,
-                value=False,
-            )
-            provides_source.sync_publish_to_topic(message_contents)
+            message_contents = [
+                {
+                    "topic_name": UpdateProvidesSourceDistroMessage.topic_name,
+                    "message_contents": {
+                        "package_name": Configuration.PACKAGE_NAME,
+                        "package_version": Configuration.PACKAGE_VERSION,
+                        "index_url": Configuration.PACKAGE_INDEX,
+                        "value": False,
+                    },
+                }
+            ]
+            with open(MESSAGE_LOCATION, "w") as f:
+                content = json.dumps(message_contents, indent=4)
+                f.write(content)
+            raise Exception("Missing source distro (message being sent)")
         else:
-            message_contents = missing_version.MessageContents(
-                package_name=Configuration.PACKAGE_NAME,
-                package_version=Configuration.PACKAGE_VERSION,
-                index_url=Configuration.PACKAGE_INDEX,
-            )
-            missing_version.sync_publish_to_topic(message_contents)
+            message_contents = [
+                {
+                    "topic_name": MissingVersionMessage.topic_name,
+                    "message_contents": {
+                        "package_name": Configuration.PACKAGE_NAME,
+                        "package_version": Configuration.PACKAGE_VERSION,
+                        "index_url": Configuration.PACKAGE_INDEX,
+                    },
+                }
+            ]
+            with open(MESSAGE_LOCATION, "w") as f:
```

Comment:
If the output is expected from the workflow task architecture we should output a default empty list when no messages need to be sent, otherwise the task should fail missing output

Reply:
Yeah, I can do that.  I'm not sure when `argo` tries to read parameters, technically the param is not required when the downloading succeeds but it might try to fill the value early :shrug:

Reply:
We experienced this issue in solver workflow and adding a default output parameter avoided errors. Thanks!